src/map/utils/battleutils.cpp

```diff
@@ -3729,28 +3729,29 @@ namespace battleutils
                 PSCEffect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_SKILLCHAIN, 0);
             }
             // Previous effect exists
-            else if (PSCEffect && PSCEffect->GetTier() == 0)
+            else if (PSCEffect->GetStartTime() + 3s < server_clock::now())
```

Comment:
Can SCEffect be nullptr here?

Reply:
there's something else to consider here. we _perceive_ a 3 second delay on retail, but there's lag time between the player and Japan. Actually, is it 3 seconds? I've not timed it. I'm unsure if BG accounts for this but gives a very vague timeframe on its skillchains page near the top that def sounds like nobody actually documented real world tests. Potentially animation times of the involved weaponskills matter. I'd have never thought something so fundamental to the game would make me say "I don't know" but here I am, doubting myself.

Reply:
Sounds like a good candidate for a setting, but also, 2.4s is very suspiciously nearby

Reply:
This is in an else of an else branch. To reach the first else branch, at least one of `PCSEffect` and `PCBEffect` must not be `nullptr`. To reach the second branch, `PCBEffect` must be `nullptr`. As such, it should impossible  to reach this branch with `PCSEffect` being `nullptr`. That said, the code might look nicer with a `nullptr` check anyway.

Reply:
small clarification: it was 3 before the contributor got here, so I don't mean we need to check it before merging this. they are just making the timing it already said actually work when it wasn't. but we should in the future confirm the timing was even right.

Reply:
I'm on my phone, I can see the upper check now, thanks!

Reply:
The SC delay should be relatively easy to test, including if it's affected by the weaponskill and/or skillchain duration.

Reply:
you can't trust the unmodified client for the timing based on the log message for skillchains. some of them (like Transfixion/fragmentation) emit the message early, and others a bit later than most and you can screw up the skillchain timing. Better testing would need an addon to see exactly when the packet came through

this will get checked in the Great Rewrite:tm:

Reply:
> Sounds like a good candidate for a setting, but also, 2.4s is very suspiciously nearby

as it happens time to japan is about 300ms. 300 there, 300 back. 600ms +2.4 seconds = 3 seconds!

Reply:
@WinterSolstice8 Well yeah. I had exactly that in mind, since I do a lot of packet sniffing. That said, I suppose my definition of "easy to test" isn't exactly standard, so point.

Reply:
I'd basically just use a script to control the input timings of the WSs involved, choosing some that have a low animation time. Then adjust the wait period down until the SC triggers about half the time.

So like, `//input /ws "Shining Strike" <t>; wait 2.9; send myalt input /ws "Gust Slash" <t>;`. Where I keep adjusting that 2.9 to find the sweet spot. Then switch to a different set of WSs to see if the window differs.